clang/test/dpct/dnn/rnn.cu

```diff
@@ -0,0 +1,355 @@
+    // CHECK: handle.rnn_get_scratchpad_workspace_size(rnnDesc, dnnl::prop_kind::forward_training, xDesc, &workSpaceSize, &reserveSpaceSize);
+    cudnnGetRNNTempSpaceSizes(handle, 
+        rnnDesc, 
+        CUDNN_FWD_MODE_TRAINING,
+        xDesc, 
+        &workSpaceSize,
+        &reserveSpaceSize
+    );
+
+    float *xData, *yData, *hxData, *hyData, *cxData, *cyData, *weightsData, *workSpaceData, *reserveSpaceData;
+    float *dxData, *dyData, *dhxData, *dhyData, *dcxData, *dcyData, *dweightsData;
+    cudnnWgradMode_t WGMode = CUDNN_WGRAD_MODE_ADD;
+    int *seqlenarray;
+    // CHECK: auto e = (handle.async_rnn_forward(rnnDesc, dnnl::prop_kind::forward_training, xDesc, xData, yDesc, yData, hDesc, hxData, hyData, cDesc, cxData, cyData, weightsSpaceSize, weightsData, workSpaceSize, workSpaceData, reserveSpaceSize, reserveSpaceData), 0);
+    auto e = cudnnRNNForward(
+        handle, 
+        rnnDesc, 
+        CUDNN_FWD_MODE_TRAINING, 
+        seqlenarray, 
+        xDesc, 
+        xData, 
+        yDesc, 
+        yData, 
+        hDesc, 
+        hxData, 
+        hyData, 
+        cDesc, 
+        cxData,
+        cyData,
+        weightsSpaceSize, 
+        weightsData, 
+        workSpaceSize, 
+        workSpaceData, 
+        reserveSpaceSize, 
+        reserveSpaceData
+    );
+// CHECK: /*
+// CHECK: DPCT1003:{{[0-9]+}}: Migrated API does not return error code. (*, 0) is inserted. You may need to rewrite this code.
+// CHECK: */
+// CHECK: e = (handle.async_rnn_backward(rnnDesc, yDesc, yData, dyData, xDesc, xData, dxData, hDesc, hxData, dhyData, dhxData, cDesc, cxData, dcyData, dcxData, weightsSpaceSize, weightsData, dweightsData, workSpaceSize, workSpaceData, reserveSpaceSize, reserveSpaceData), 0);
+    e = cudnnRNNBackwardData_v8(
+        handle,
+        rnnDesc,
+        seqlenarray,
+        yDesc,
+        yData,
+        dyData,
+        xDesc,
+        dxData,
+        hDesc,
+        hxData,
+        dhyData,
+        dhxData,
+        cDesc,
+        cxData,
+        dcyData,
+        dcxData,
+        weightsSpaceSize,
+        weightsData,
+        workSpaceSize, 
+        workSpaceData, 
+        reserveSpaceSize, 
+        reserveSpaceData
+    );
+// CHECK: /*
+// CHECK: DPCT1027:{{[0-9]+}}: The call to cudnnRNNBackwardWeights_v8 was replaced with 0 because this call and cudnnRNNBackwardData_v8 are migrated to a single function call async_rnn_backward
+// CHECK: */
+// CHECK: e = 0;
+    e = cudnnRNNBackwardWeights_v8(
+        handle,
+        rnnDesc,
+        WGMode,
+        seqlenarray,
+        xDesc,
+        xData,
+        hDesc,
+        hxData,
+        yDesc,
+        yData,
+        weightsSpaceSize,
+        dweightsData,
+        workSpaceSize, 
+        workSpaceData, 
+        reserveSpaceSize, 
+        reserveSpaceData
+    );
+
+   float *y1Data;
+// CHECK:     /*
+// CHECK:     DPCT1007:{{[0-9]+}}: Migration of cudnnRNNBackwardData_v8 is not supported.
```

Comment:
Consider to introduce a new and more speicfic warnning for this scenario. as here for Line200,  we know the reason why it can not be migrated, and for some other case like Line177, the tool can support migration of cudnnRNNBackwardData_v8.

Reply:
Ok, I'll implement this request in another separate PR.